server/services/store/sqlstore/migrations/migrations_files/000015_add_teams_and_boards.up.sql

```diff
@@ -0,0 +1,361 @@
+                 fields->'icon', (fields->'showDescription')::text::boolean, (fields->'isTemplate')::text::boolean,
+                 '{}', fields->'cardProperties', fields->'columnCalculations', create_at,
+                 update_at, delete_at
+          FROM {{.prefix}}blocks
+          WHERE type='board'
+  );
+  INSERT INTO {{.prefix}}boards_history (
+      SELECT id, insert_at, '0', channel_id, created_by, modified_by, 'O', title, (fields->'description')::text,
+                 fields->'icon', (fields->'showDescription')::text::boolean, (fields->'isTemplate')::text::boolean,
+                 '{}', fields->'cardProperties', fields->'columnCalculations', create_at,
+                 update_at, delete_at
+          FROM {{.prefix}}blocks_history
+          WHERE type='board'
+  );
+  {{end}}
+  {{if .mysql}}
+  INSERT INTO {{.prefix}}boards (
+      SELECT id, insert_at, '0', channel_id, created_by, modified_by, 'O', title, fields->'$.description',
+                 fields->'$.icon', fields->'$.showDescription', fields->'$.isTemplate',
+                 '{}', fields->'$.cardProperties', fields->'$.columnCalculations', create_at,
+                 update_at, delete_at
+          FROM {{.prefix}}blocks
+          WHERE type='board'
+  );
+  INSERT INTO {{.prefix}}boards_history (
+      SELECT id, insert_at, '0', channel_id, created_by, modified_by, 'O', title, fields->'$.description',
+                 fields->'$.icon', fields->'$.showDescription', fields->'$.isTemplate',
+                 '{}', fields->'$.cardProperties', fields->'$.columnCalculations', create_at,
+                 update_at, delete_at
+          FROM {{.prefix}}blocks_history
+          WHERE type='board'
+  );
+  {{end}}
+  {{if .sqlite}}
+  INSERT INTO {{.prefix}}boards
+      SELECT id, insert_at, '0', channel_id, created_by, modified_by, 'O', title, json_extract(fields, '$.description'),
+                 json_extract(fields, '$.icon'), json_extract(fields, '$.showDescription'), json_extract(fields, '$.isTemplate'),
+                 '{}', json_extract(fields, '$.cardProperties'), json_extract(fields, '$.columnCalculations'), create_at,
+                 update_at, delete_at
+          FROM {{.prefix}}blocks
+          WHERE type='board'
+  ;
+  INSERT INTO {{.prefix}}boards_history
+      SELECT id, insert_at, '0', channel_id, created_by, modified_by, 'O', title, json_extract(fields, '$.description'),
+                 json_extract(fields, '$.icon'), json_extract(fields, '$.showDescription'), json_extract(fields, '$.isTemplate'),
+                 '{}', json_extract(fields, '$.cardProperties'), json_extract(fields, '$.columnCalculations'), create_at,
+                 update_at, delete_at
+          FROM {{.prefix}}blocks_history
+          WHERE type='board'
+  ;
+  {{end}}
+{{end}}
+
+
+{{- /* Update block references to boards*/ -}}
+{{if .sqlite}}
+  UPDATE {{.prefix}}blocks as B
```

Comment:
nit: Would this be a much faster migration -

```
UPDATE {{.prefix}}blocks as B
SET board_id = parent_id
WHERE type = 'card' OR type = 'view';
```

I think the subqueries will take a lot of time.

Reply:
The goal of the query is to have all blocks belonging to a board, and that would leave the inner blocks out sadly.